### PR TITLE
Record User stamp

### PR DESCRIPTION
**Pull request by StupidCodeFactory**
Make sure https://github.com/ministryofjustice/prison-visits-2/pull/1701 is merged first

**CHANGES:**

Sends the current staff user's email to NOMIS API when booking a visit.

### CODE REVIEW

File: app/services/create_nomis_visit.rb [hunk 1]
```diff
@@ -71,7 +72,7 @@ def booking_params
       override_visitor_restrictions: false,
       override_vo_balance: false,
       override_slot_capacity: false,
-      client_unique_ref: visit.id
+      client_unique_ref: visit.id, comment: creator.email
```
Comment (TobyRet) [inline app/services/create_nomis_visit.rb]: new line

Comment (TobyRet) [reply app/services/create_nomis_visit.rb]: creator.email needs to be passed to NOMIS via pvb_user_id not comment

Comment (StupidCodeFactory) [reply app/services/create_nomis_visit.rb]: @TobyRet where do you see that pvb_user_id field in API docs?
I think it needs to be passed as through the `X-PVB-User-id` header.
https://github.com/ministryofjustice/nomis-api/blob/dev/src/main/java/syscon/nomisapi/visit/OffenderVisits.java#L628

File: app/services/booking_responder/accept.rb [hunk 2]
```diff
@@ -29,7 +29,9 @@ def book_to_nomis
     end
 
     def nomis_visit_creator
-      @nomis_visit_creator ||= CreateNomisVisit.new(visit)
+      @nomis_visit_creator ||= CreateNomisVisit.new(
+        visit, creator: staff_response.creator
```
Comment (TobyRet) [inline app/services/booking_responder/accept.rb]: Is it also worth passing the visit as a hash for consistency / increased safety?

**Comment by StupidCodeFactory** [comment]
@TobyRet @KPobeeNorris I've reopened the PR against master here https://github.com/ministryofjustice/prison-visits-2/pull/1723

so I'm closing this

File: app/services/booking_responder/accept.rb (hunk 2 above)
Comment (StupidCodeFactory) [reply app/services/booking_responder/accept.rb]: No, using keywords argument everywhere is just insane clutter. Here `visit` is the main subject of the class which also bares 'Visit` in the class name, I'd rather remove the `creator` keyword argument

**Pull request closed without merging**